### ISSUE TITLE
bzip2: source tarball not hosted anymore

### ISSUE THREAD

**Issue by gweodoo** [spack/spack#8963]
### EDIT
Just discovered a duplicate from #8924. Can be closed

### TL;DR 
The URL used by bzip2 recipe, `http://www.bzip.org`, seems not available anymore and now, Spack is not able to install it nor any package depending on it (Git in my case, required by the Gettext package).

-----

I have no idea whether it is temporary or not. In the meantime, it is not possible to install `bzip2`from Spack since the URL `bzip.org` does not refers to the actual project. A website is still reachable at this URL.

This is not a issue with Spack but, in the meantime, it is not possible to install any package depending on it.
I could set up a Github repo & tarballs but I think someone probably hosts one somewhere.

### Expected Result

Install `bzip2`properly.

### Actual Result

It did not install, obviously, with the following output:
```console
$ spack install bzip2
==> Installing bzip2
==> Fetching http://www.bzip.org/1.0.6/bzip2-1.0.6.tar.gz
######################################################################### 100,0%
==> Warning: The contents of
  the archive
   look like HTML.
  The checksum will likely be bad.  If it is, you can use
  'spack clean <package>' to remove the bad archive, then
  fix your internet gateway issue and install again.
==> Error: ChecksumError: md5 checksum failed for /spack/var/spack/stage/bzip2-1.0.6-lm4lp2bp22i2fmz3amqd2stay22wq6oh/bzip2-1.0.6.tar.gz
    Expected 00b516f4704d4a7cb50a1d97e6e8e15b but got ef1d6942767c9caca49f51f2d5cbf931
...
```

### Steps to reproduce the issue

```console
$ spack install bzip2
```

### Information on your system

```console
$ uname -r
4.17.3-200.fc28.x86_64
```

And absolutely no exotic configuration at all.
 1. which platform you are using
 2. any relevant configuration detail (custom `packages.yaml` or `modules.yaml`, etc.)

**Comment by vsoch** [comment]
I just hit this issue, but with a different mirror, see #12749

**Comment by vsoch** [comment]
Just kidding, my branch wasn't updated, the fix was already done! Sorry about that.
